code/game/atoms.dm

```diff
@@ -74,16 +76,20 @@
 	if(color)
 		add_atom_colour(color, FIXED_COLOUR_PRIORITY)
 
-	if (light_power && light_range)
+	if(light_power && light_range)
 		update_light()
 
-	if (opacity && isturf(loc))
+	if(opacity && isturf(loc))
 		var/turf/T = loc
 		T.has_opaque_atom = TRUE // No need to recalculate it in this case, it's guaranteed to be on afterwards anyways.
 
-	if (canSmoothWith)
+	if(canSmoothWith)
 		canSmoothWith = typelist("canSmoothWith", canSmoothWith)
 
+	if(datum_outputs)
+		for(var/A in datum_outputs)
+			A = SSoutputs.outputs[A]
+

```

Comment:
You set the var A to the actual datum output based on type of the output but then do nothing with it
You do no actually rewrite anything in the list

Reply:
i actually did, i just moved the cooldown reset and it works now